web/src/buddy_data.js

```diff
@@ -66,7 +55,27 @@ export function level(user_id) {
     }
 }
 
-export function compare_function(a, b) {
+export function would_receive_message(user_id, current_sub, pm_ids_string) {
```

Comment:
from Tim earlier:

> Well you probably want a different name then -- something like `user_matches_narrow` or something -- since the name `would_receive_message` was chosen to refer to specifically if that user would receive the message currently being composed.
> 
> I also note that this has an `ids.includes` check in it -- some care may be needed here to use sets and thus avoid accidentally quadratic code.